pkg/promql/promql.go

```diff
@@ -21,6 +23,7 @@ func SetLabelMatchers(query string, labelMatchers []LabelMatcher) string {
 func LabelsSetPromQL(query, labelMatchType, name, value string) string {
 	expr, err := parser.ParseExpr(query)
 	if err != nil {
+		fmt.Println("Error parsing query:", err)
```

Comment:
I added this, as in some cases like adding $__rate_interval to your query, ensures that you fail parsing here which ends up being blank query in YAML.

Reply:
Also this utility adds the matcher to all parts of the query. But for certain ones, you really need the matcher in only one part

Reply:
Plan to address this in next PR!